druid/src/widget/split.rs

```diff
@@ -0,0 +1,327 @@
+};
+
+///A container containing two other widgets, splitting the area either horizontally or vertically.
+pub struct Split<T: Data> {
+    split_point: f64,
+    split_direction: Axis,
+    splitter_size: f64, // Should this be a u32?
+    draggable: bool,
+    child1: WidgetPod<T, Box<dyn Widget<T>>>,
+    child2: WidgetPod<T, Box<dyn Widget<T>>>,
+}
+
+/// A builder for a widget which divides space horizontally
+/// The splitter bar will run top-to-bottom
+///
+/// The actual widget it implemented by [`Split`], but this builder
+/// is provided for convenience.
+///
+/// [`Split`]: struct.Split.html
+pub struct HorizontalSplit;
+impl HorizontalSplit {
+    /// Create a new split panel which splits the horizontal area,
+    /// leaving the full height available to both children.
+    ///
+    /// Note that `split_point` must be between 0 and 1  
+    pub fn new<T: Data>(
+        split_point: f64,
+        splitter_size: f64,
+        draggable: bool,
+        child1: impl Widget<T> + 'static,
+        child2: impl Widget<T> + 'static,
+    ) -> Split<T> {
+        Split::new(
+            Axis::Horizontal,
+            split_point,
+            splitter_size,
+            draggable,
+            child1,
+            child2,
+        )
+    }
+}
+/// A builder for a widget which divides space vertically
+/// The splitter bar will run left-to-right
+///
+/// The actual widget it implemented by [`Split`], but this builder
+/// is provided for convenience.
+///
+/// [`Split`]: struct.Split.html
+pub struct VerticalSplit;
+impl VerticalSplit {
+    /// Create a new split panel which splits the vertical area,
+    /// leaving the full width available to both children.
+    ///
+    /// Note that `split_point` must be between 0 and 1  
+    pub fn new<T: Data>(
+        split_point: f64,
+        splitter_size: f64,
+        draggable: bool,
+        child1: impl Widget<T> + 'static,
+        child2: impl Widget<T> + 'static,
+    ) -> Split<T> {
+        Split::new(
+            Axis::Vertical,
+            split_point,
+            splitter_size,
+            draggable,
+            child1,
+            child2,
+        )
+    }
+}
+
+impl<T: Data> Split<T> {
+    ///Create a new split panel.
+    ///Note that split_point must be between 0 and 1.
+    fn new(
+        split_direction: Axis,
+        split_point: f64,
+        splitter_size: f64,
+        draggable: bool,
+        child1: impl Widget<T> + 'static,
+        child2: impl Widget<T> + 'static,
+    ) -> Split<T> {
+        assert!(
+            split_point > 0.0 && split_point < 1.0,
+            "Invalid split point {} provided!",
+            split_point
+        );
+        Split {
+            split_point,
+            split_direction,
+            splitter_size,
+            draggable,
+            child1: WidgetPod::new(child1).boxed(),
+            child2: WidgetPod::new(child2).boxed(),
+        }
+    }
+    fn splitter_hit_test(&self, size: Size, mouse_pos: Point) -> bool {
```

Comment:
Something this is making me realize: we're conflating the _visual_ size of the splitter with its size for the purpose of hit-testing, and I think these should be different; or at the very least we should have a minimum size that we use for the purpose of hit-testing.

Reply:
That makes sense to me, but I do worry a bit about 'absorbing' input over the children. or both the splitter and the child reacting. Currently, I mark the mousedown handled, so the absorption case is what would happen.

I also ignore 'draggable' if the splitter size is 0.

If we do go for 'fuzzy', do we just want to add a fuzz-constant (2? 3?) to the hit-test and still mark events handled? Or something more nuanced?

Reply:
What we might want to do is check the children and see if either is active, and if they're active then we send them the event first.

Reply:
I went with the check-for-active. I'm not sure if that's the nicest way to do it. But it seemed more-sane to me? I also check for marked-resolved, and return in that case. Maybe we don't want to do that?